src/js/game/GameController.js

```diff
@@ -794,22 +794,36 @@ class GameController {
 
   moveForward(commandQueueItem) {
     const target = commandQueueItem.target;
-    const moveOffset = this.directionToOffset(target.facing);
+    let entity = this.getEntity(target);
+    if (entity === this.agent) {
+      entity.movementState = entity.facing;
```

Comment:
Why does this special logic only apply to the Agent?

Reply:
(Same question below.  In theory this should also work for a Player that is controlled by code, right?)